From 6dfbf88e09cde7dd4af94a727319ee03a74870bf Mon Sep 17 00:00:00 2001
From: Ajay Dudani <adudani@google.com>
Date: Mon, 5 Feb 2018 10:44:27 -0800
Subject: [PATCH 103/306] Add display panel low power mode global setting &
 mapping to sys property.

This will help us run P/H experiments by controlling display panel low
power mode system property through P/H.

Bug: 72121774
Test: Compile, boot taimen.
Change-Id: Id4c95bcd5616e9749f9dcc9d9f4773ffdad3e994
(cherry picked from commit bcfcd6c9bb6fc728e433c6156fdfa51432f92adf)
---
 core/java/android/provider/Settings.java                 | 9 +++++++++
 .../src/android/provider/SettingsBackupTest.java         | 1 +
 .../server/am/GlobalSettingsToPropertiesMapper.java      | 1 +
 3 files changed, 11 insertions(+)

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index f1793714993..46a97e4d8be 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -10332,6 +10332,15 @@ public final class Settings {
          */
         public static final String FPS_DEVISOR = "fps_divisor";
 
+        /**
+         * Flag to enable or disable display panel low power mode (lpm)
+         * false -> Display panel power saving mode is disabled.
+         * true  -> Display panel power saving mode is enabled.
+         *
+         * @hide
+         */
+        public static final String DISPLAY_PANEL_LPM = "display_panel_lpm";
+
         /**
          * App standby (app idle) specific settings.
          * This is encoded as a key=value list, separated by commas. Ex:
diff --git a/core/tests/coretests/src/android/provider/SettingsBackupTest.java b/core/tests/coretests/src/android/provider/SettingsBackupTest.java
index e1f95a3ca57..d462e5eac60 100644
--- a/core/tests/coretests/src/android/provider/SettingsBackupTest.java
+++ b/core/tests/coretests/src/android/provider/SettingsBackupTest.java
@@ -196,6 +196,7 @@ public class SettingsBackupTest {
                     Settings.Global.DEVICE_PROVISIONED,
                     Settings.Global.DEVICE_PROVISIONING_MOBILE_DATA_ENABLED,
                     Settings.Global.DISK_FREE_CHANGE_REPORTING_THRESHOLD,
+                    Settings.Global.DISPLAY_PANEL_LPM,
                     Settings.Global.DISPLAY_SCALING_FORCE,
                     Settings.Global.DISPLAY_SIZE_FORCED,
                     Settings.Global.DNS_RESOLVER_MAX_SAMPLES,
diff --git a/services/core/java/com/android/server/am/GlobalSettingsToPropertiesMapper.java b/services/core/java/com/android/server/am/GlobalSettingsToPropertiesMapper.java
index 9c2ee873fa6..fe659784732 100644
--- a/services/core/java/com/android/server/am/GlobalSettingsToPropertiesMapper.java
+++ b/services/core/java/com/android/server/am/GlobalSettingsToPropertiesMapper.java
@@ -42,6 +42,7 @@ class GlobalSettingsToPropertiesMapper {
         {Settings.Global.SYS_VDSO, "sys.vdso"},
         {Settings.Global.FPS_DEVISOR, ThreadedRenderer.DEBUG_FPS_DIVISOR},
         {Settings.Global.UID_CPUPOWER, "uid.cpupower"},
+        {Settings.Global.DISPLAY_PANEL_LPM, "sys.display_panel_lpm"},
     };
 
 
-- 
2.17.1

